### ISSUE TITLE
html-report plugin fails during test execution with proto message failure

### ISSUE THREAD

**Issue by alpha1592** [getgauge/gauge#1350]
### Expected behavior
Html-report plugin should complete generation of report every time tests are executed. 

### Actual behavior

html-report plugin fails with repeating error messages related to proto message. This is an intermittent issue meaning that sometimes the test execution passes and sometimes it fails. I am running roughly 35 test specifications with 1,273 test scenarios among them. This happens both in single stream mode or in parallel streams (inParallel=true). I have never seen this issue for individual tests run thru IntelliJ or running one spec from command line. This happens when we run the test suite of all 35 specs. The tests that i am running are soap based web services testing and therefore, i write XML message to the HTML report so we can see the contents of the request/response messages. I use **`Gauge.writeMessage(xmlPayload)`** to write payload which may contain chars such as |, ^, & and others. Could this issue be happening as a result of un-escaped special characters like &lt; &gt; "" &amp; etc? The spec files are also quite large in some cases (over 10k lines). I also use 4 hash signs to define a "sub-step" for my tests. While its not supported by Gauge, it does show up nicely  formatted as H4 in documentation (spectacle) and html reports. There doesnt seem to be any clear reason for the failure and i dont know where to look to get more details. The html-report plugin fails at different tests/points in the execution during each execution. 

Both api.log and gauge.log do not provide any errors. The logs below are the java project log files as seen in console. 

### Logs (snippet)
```
[runner: 2] [INFO] 	#### ART-0036-AP03-EP38-17-ORU-C
[runner: 6] [ERROR] ART-0036-AP01-EP17-37-ORM-A Failed. 
[runner: 6] [INFO] 	#### ART-0036-AP01-EP17-38-SPQ-Z02
[runner: 5] [INFO] 	#### ART-0031-AP03-EP22-12-SPQ-Z02
[html-report] 2019/02/25 11:37:45 Failed to read proto message: proto: can't skip unknown wire type 6
[html-report] 2019/02/25 11:37:45 Failed to read proto message: proto: can't skip unknown wire type 6
[html-report] 2019/02/25 11:37:45 Failed to read proto message: proto: can't skip unknown wire type 6
[html-report] 2019/02/25 11:37:45 Failed to read proto message: proto: can't skip unknown wire type 6
[html-report] 2019/02/25 11:37:45 Failed to read proto message: proto: can't skip unknown wire type 6
..... continues forever
..... Have to kill gauge.exe from task manager. 
```
### Another Log snippet
```
[runner: 5]   ## ART_0045_AP03_EP05
[runner: 5] [INFO] 	#### ART-0045-AP03-EP05-01-ORU-C
[runner: 4] [INFO] 	#### ART-0042-BP-EP05-02-SPQ-Z02
[runner: 4] [INFO] 	#### ART-0042-BP-EP05-03-ORM-C
[runner: 1] [INFO] 	#### a_ART-0050-AP04-EP00_1-20-SPQ-Z01
[runner: 1] [INFO] 	#### a_ART-0050-AP04-EP00_1-21-SPQ-Z01
[runner: 6] [INFO] 	#### ART-0044-AP03-EP09-08-SPQ-Z02
[html-report] 2019/02/25 13:00:54 Failed to read proto message: unexpected EOF
[html-report] 2019/02/25 13:00:54 Failed to read proto message: unexpected EOF
[html-report] 2019/02/25 13:00:54 Failed to read proto message: unexpected EOF
[html-report] 2019/02/25 13:00:54 Failed to read proto message: unexpected EOF
[html-report] 2019/02/25 13:00:54 Failed to read proto message: unexpected EOF
..... continues forever
..... Have to kill gauge.exe from task manager. 

```

### Steps to reproduce
unfortunately, due to privacy issues, i cannot provide the actual test specs / code. I will try to create a new project and try to reproduce the issue. 

1. Using Gauge with Maven + Java  
2. Run mvn gauge:execute -DspecsDir=specs/ -DinParallel=true
3. Some time during the execution, the HTML-REPORT plugin will fail. 

### Sample Spec File
```
# RQ-AR-0001
Add a description for this Test Specification
tags: regression, r1.00

## Scenario 1
Add a test scenario description here

#### Sub-Scenario 1
 * Send "MessageA" message to Soap-Service using prefix "A" and certificate "certificate1"
 * Validate that the response message contains "Some|Ack|"
 * Clean up and Finish

#### Sub-Scenario 2
 * Send "MessageB" message to Soap-Service using prefix "B" and certificate "certificate2"
 * Validate that the response message contains "Some|Ack|"
 * Clean up and Finish
```

### Gauge version
```
Gauge version: 1.0.4
Commit Hash: 3a9a647

Plugins
-------
flash (0.0.1)
html-report (4.0.7)
java (0.7.1)
json-report (0.2.2)
screenshot (0.0.1)
spectacle (0.1.3)
xml-report (0.2.1)

```

### Manifest.json
```
{
  "Language": "java",
  "Plugins": [
    "html-report",
    "flash"
  ]
}
```

### Java Version
```
java version "1.8.0_191"
Java(TM) SE Runtime Environment (build 1.8.0_191-b12)
Java HotSpot(TM) 64-Bit Server VM (build 25.191-b12, mixed mode)
```

**Comment by Debashis9012** [comment]
Duplicate of #https://github.com/getgauge/html-report/issues/213
